frontend/src/components/maps/MinePin.js

```diff
@@ -126,22 +106,22 @@ export class MinePin extends Component {
           content: "{templateContent}",
         });
 
-        const fclData = mineIds.reduce((result, id) => {
-          const point = this.points(id);
-          if (!point) {
-            return result;
+        const mapPopupString = renderToString(<MapPopup id="{mineId}" />);
+
+        // The previous code processing point data was nicely seperated into functions, but slow with
+        // large datasets taking 5-10 seconds for 50000 points. The code below is ~50ms for 50000 points
+        // and uses significantly lower memory.
+        const fclData = [];
```

Comment:
🙏 💯 🔥 
Makes me almost wonder if we should refactor all uses of `.reduce` in our frontend given all our datasets are relatively large